### PR TITLE
translation macro placeline switch: apply the french geonames value also for AP ingested stories [SDCP-560]

### PR DESCRIPTION
**Pull request by GyanP**

### CODE REVIEW

**Comment by petrjasek** [comment]
test pls

File: server/cp/ingest/parser/ap.py [hunk 1]
```diff
@@ -191,7 +202,31 @@ def parse(self, data, provider=None):
...
+                    if float(item_["lat"]) == lat and float(item_["lng"]) == lon:
+                        formatted_item = format_geoname_item(item_)
+                        break
+
+                if formatted_item:
+                    item["dateline"]["located"].update(formatted_item)
+                    # set place key required while translation
+                    item["dateline"]["located"]["place"] = formatted_item
+
+            except Exception:
```
Comment (petrjasek) [inline server/cp/ingest/parser/ap.py:229]: pls use more specific exception

File: server/cp/ingest/parser/ap.py [hunk 2]
```diff
@@ -191,7 +202,31 @@ def parse(self, data, provider=None):
...
+                    ("style", app.config["GEONAMES_SEARCH_STYLE"]),
+                ]
+                for feature_class in app.config["GEONAMES_FEATURE_CLASSES"]:
+                    params.append(("featureClass", feature_class.upper()))
+
+                # get geo data from geoname_request
+                json_data = geonames_request("search", params)
+
+                formatted_item = None
+                for item_ in json_data.get("geonames", []):
```
Comment (petrjasek) [inline server/cp/ingest/parser/ap.py:219]: would be good to use some other name like place

File: server/tests/ingest/parser/ap_test.py [hunk 3]
```diff
@@ -115,10 +124,13 @@ def test_parse(self):
...
         self.assertIn("located", dateline)
         self.assertEqual("Cheyenne", dateline["located"]["city"])
+        self.assertEqual("5821086", dateline["located"]["code"])
+        self.assertEqual("America/Denver", dateline["located"]["tz"])
         self.assertEqual("Wyoming", dateline["located"]["state"])
         self.assertEqual("WY", dateline["located"]["state_code"])
         self.assertEqual("United States", dateline["located"]["country"])
-        self.assertEqual("USA", dateline["located"]["country_code"])
+        self.assertEqual("US", dateline["located"]["country_code"])
+        self.assertIn("place", dateline["located"])
```
Comment (petrjasek) [inline server/tests/ingest/parser/ap_test.py:133]: can you also test contents of place, this would work with `place=None`

**Comment by petrjasek** [comment]
I think it might be better to handle this when translating the item, otherwise this is fetching it for every ap item which is thousands per day but there will be only a few translations made at the end, geonames api is paid service so this seems like wasting resources
